RD-2108 Tests for markers in Deployments Map

Gelio opened the pull request:
The PR implemented tests for the map functionalities added in #1325:
1. The markers for deployments are displayed
2. The marker images are based on deployment statuses
3. The marker popups contain deployment names

## System tests

I'm only running the `deployments_view_spec.ts` suite because that is the only suite that is being modified by this PR. No other suites are affected, AFAIK.

https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/500/pipeline

Ran 2021-04-23 16:00 CEST

Gelio commented on test/cypress/integration/widgets/deployments_view_spec.ts:
```diff
@@ -81,7 +81,6 @@ describe('Deployments View widget', () => {
         cy.get('.widget').filter('.deploymentsViewWidget, .deploymentsViewDrilledDownWidget').find('.widgetItem');
     const getDeploymentsViewTable = () => getDeploymentsViewWidget().get('.gridTable');
     const getDeploymentsViewDetailsPane = () => getDeploymentsViewWidget().get('.detailsPane');
-    // TODO(RD-2090): use a better selector for the map
```
I did not find a better selector. The map is a UI-heavy element, and looks hard to select accessibly :slightly_smiling_face:

Gelio commented:
FYI, the system tests for `deployments_view_spec` passed :slightly_smiling_face:

kubama commented on test/cypress/integration/widgets/deployments_view_spec.ts:
```diff
@@ -500,13 +502,77 @@ describe('Deployments View widget', () => {
+            });
+
+            getDeploymentsViewMap().within(() => {
+                cy.get('.leaflet-marker-icon').should('have.length', 3).and('be.visible');
```
Really cosmetic, but I think it reads better if assertions are done in the order starting from general and ending on detailed.
```suggestion
                cy.get('.leaflet-marker-icon').should('be.visible').and('have.length', 3);
```

Gelio replied:
Good idea :+1: Thanks

When writing the code, I first wrote the `have.length` one and then added `be.visible` several minutes later, so I did not read the whole line :smile: Makes sense, I'll apply it soon

Gelio replied:
Done

Gelio commented:
I have rebased the PR on the base branch since it has changed, and I have addressed the comments :slightly_smiling_face: I'll wait for #1325 to be merged before rebasing again and asking for final approvals

I'll not be rerunning system tests, since the applied change was a reorder of assertions

kubama reviewed (approved):
Thanks for catching and addressing the `only` issue 🙂.
